python/setup.py

```diff
@@ -0,0 +1,92 @@
+import os
+import sys
+import subprocess
+
+import setuptools
+import setuptools.command.build_py
+import setuptools.command.install
+import setuptools.command.develop
+import setuptools.command.test
+
+_this_dir = os.path.dirname(os.path.abspath(__file__))
+_clib_dir = os.path.abspath(os.path.join(_this_dir, 'opencc', 'clib'))
+_opencc_rootdir = os.path.abspath(os.path.join(_this_dir, '..'))
+
+assert os.path.isfile(os.path.join(_opencc_rootdir, 'CMakeLists.txt'))
+
+_build_dir = build_dir = os.path.join(_opencc_rootdir, 'build', 'python')
+_libopenccfile = os.path.join(_clib_dir, 'lib', 'libopencc.so')
+
+
+def _build_libopencc():
+    print('building libopencc')
+    assert subprocess.call('command -v make', shell=True) == 0, \
+        'Requires make'
+    assert subprocess.call('command -v cmake', shell=True) == 0, \
+        'Requires cmake'
+    # Probably also needs to check for c-compilier
+
+    errno = subprocess.call((
+        'mkdir -p {build_dir};'
+        'cd {build_dir};'
+        'cmake '
+        '-DBUILD_DOCUMENTATION:BOOL=OFF '
+        '-DENABLE_GTEST:BOOL=OFF '
+        '-DCMAKE_BUILD_TYPE=Release '
+        '-DCMAKE_INSTALL_PREFIX={clib_dir} '
+        '../..;'
+        'make -j;'
+        'make install;'
+    ).format(
+        build_dir=_build_dir,
+        clib_dir=_clib_dir
+    ), shell=True)
+
+    assert errno == 0, 'Build failed'
+    assert os.path.isfile(_libopenccfile)
+
+
+class BuildPyCommand(setuptools.command.build_py.build_py):
+    def run(self):
+        _build_libopencc()
+        super().run()
+
+
+class InstallCommand(setuptools.command.install.install):
+    def run(self):
+        _build_libopencc()
+        super().run()
+
+
+class DevelopCommand(setuptools.command.develop.develop):
+    def run(self):
+        _build_libopencc()
+        super().run()
+
+
+class PyTestCommand(setuptools.command.test.test):
+    def run_tests(self):
+        import pytest
+        errno = pytest.main([])
+        sys.exit(errno)
+
+
+setuptools.setup(
+    name='opencc',
+    version='0.0.1',
```

Comment:
Can this version be the same as opencc version? It would be better if you can read it from CMakelist.txt

Reply:
This is something I originally intended to do but one worry is that packaging might be problematic. I don't have too much experience with PyPi but there might be the possibility that CMakelist.txt might not be findable for PyPi because it is located in the parent directory to `setup.py`.

I'll run a few experiments and get back to you on that.

Reply:
Okay I've checked and my worries weren't necessary.

`version` will now be parsed directly from `CMakeLists.txt` using regex.
A `version.py` file will be generated on build time to add the `__version__` variable to the package.

Reply:
Fixed in 166797f